Makefile

```diff
@@ -128,3 +132,29 @@ _topolvm_delete:
 	DEVICE_NAME="$$(sudo losetup -j "${LVM_DISK}" | cut -d: -f1)" && \
 	[ -n "$${DEVICE_NAME}" ] && sudo losetup -d $${DEVICE_NAME} || true
 	sudo rm -rf "$$(dirname "${LVM_DISK}")" || true
+
+# When run inside a container, the file contents are redirected via stdin and
+# the output of errors does not contain the file path. Work around this issue
+# by replacing the '^-:' token in the output by the actual file name.
+.PHONY: _hadolint
+_hadolint:
+	set -euo pipefail && \
+	RET=0 && \
+	FILES=$$(find . -iname '*containerfile*' -o -iname '*dockerfile*' | grep -v "vendor\|_output\|origin\|.git") && \
+	for f in $${FILES} ; do \
+    	echo "$${f}" ; \
+    	if ! podman run --rm -i \
+        		-v "$(CURDIR)/.hadolint.yaml:/.hadolint.yaml:Z" \
+        		ghcr.io/hadolint/hadolint:2.12.0 < "$${f}" | sed "s|^-:|$${f}:|" ; then \
+			RET=1 ; \
+		fi ; \
+	done ; \
+	exit $${RET}
+
+.PHONY: _shellcheck
+_shellcheck:
+	shopt -s globstar nullglob && \
+	podman run --rm -i \
+		-v "$(CURDIR):/mnt:Z" \
+		docker.io/koalaman/shellcheck:stable --format=gcc --external-sources \
```

Comment:
Should we pin shellcheck to a version too? I think its safer, just like hadolint.